cmd/bosun/sched/template.go

```diff
@@ -230,78 +232,96 @@ func (c *Context) eval(v interface{}, filter bool, series bool, autods int) (res
 	default:
 		return nil, "", fmt.Errorf("expected string, expression or resultslice, got %T (%v)", v, v)
 	}
-	if filter {
+	if filter && err != nil {
```

Comment:
will fix, the shared err return is confusing anyways. As proved by the existing flaw in logic that was already there with shadowing lin 233.